File: javav2/example_code/translate/src/main/java/com/example/translate/DescribeTextTranslationJob.java
//snippet-sourcedescription:[DescribeTextTranslationJob.kt demonstrates how to describe a translation job.]
//snippet-keyword:[AWS SDK for Kotlin]
//snippet-keyword:[Code Sample]
//snippet-service:[Amazon Translate]
//snippet-sourcetype:[full-example]
//snippet-sourcedate:[05/19/2022]

/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/

package com.example.translate;

// snippet-start:[translate.java2._describe_jobs.import]
import software.amazon.awssdk.auth.credentials.ProfileCredentialsProvider;
import software.amazon.awssdk.regions.Region;
import software.amazon.awssdk.services.translate.TranslateClient;
import software.amazon.awssdk.services.translate.model.DescribeTextTranslationJobRequest;
import software.amazon.awssdk.services.translate.model.DescribeTextTranslationJobResponse;
import software.amazon.awssdk.services.translate.model.TranslateException;
// snippet-end:[translate.java2._describe_jobs.import]

/**
 * Before running this Java V2 code example, set up your development environment, including your credentials.
 *
 * For more information, see the following documentation topic:
 *
 * https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/get-started.html
 */
public class DescribeTextTranslationJob {

    public static void main(String[] args) {


        final String usage = "\n" +
                "Usage:\n" +
                "    <id> \n\n" +
                "Where:\n" +
                "    id - A translation job ID value. You can obtain this value from the BatchTranslation example.\n";

        if (args.length != 1) {
            System.out.println(usage);
            System.exit(1);
        }

        String id = args[0];
        Region region = Region.US_WEST_2;
        TranslateClient translateClient = TranslateClient.builder()
                .region(region)
                .credentialsProvider(ProfileCredentialsProvider.create())
                .build();

        describeTextTranslationJob(translateClient, id);
        translateClient.close();
    }

    // snippet-start:[translate.java2._describe_jobs.main]
    public static void describeTextTranslationJob(TranslateClient translateClient, String id) {

      try {

        DescribeTextTranslationJobRequest textTranslationJobRequest = DescribeTextTranslationJobRequest.builder()
                .jobId(id)
                 .build();

        DescribeTextTranslationJobResponse jobResponse = translateClient.describeTextTranslationJob(textTranslationJobRequest);
        System.out.println("The job status is "+jobResponse.textTranslationJobProperties().jobStatus());
        System.out.println("The source language is "+jobResponse.textTranslationJobProperties().sourceLanguageCode());
        System.out.println("The target language is "+jobResponse.textTranslationJobProperties().targetLanguageCodes());

      } catch (TranslateException e) {
          System.err.println(e.getMessage());
          System.exit(1);
      }
    }
    // snippet-end:[translate.java2._describe_jobs.main]
}

File: kotlin/services/translate/src/main/kotlin/com/kotlin/translate/DescribeTextTranslationJob.kt
// snippet-sourcedescription:[DescribeTextTranslationJob.kt demonstrates how to describe a translation job.]
// snippet-keyword:[AWS SDK for Kotlin]
// snippet-service:[Amazon Translate]

/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/

package com.kotlin.translate

// snippet-start:[translate.kotlin._describe_jobs.import]
import aws.sdk.kotlin.services.translate.TranslateClient
import aws.sdk.kotlin.services.translate.model.DescribeTextTranslationJobRequest
import kotlin.system.exitProcess
// snippet-end:[translate.kotlin._describe_jobs.import]

/**
Before running this Kotlin code example, set up your development environment,
including your credentials.

For more information, see the following documentation topic:
https://docs.aws.amazon.com/sdk-for-kotlin/latest/developer-guide/setup.html
 */
suspend fun main(args: Array<String>) {

    val usage = """
    Usage:
        <jobId> 

    Where:
         jobId - A translation job ID value. You can obtain this value from the BatchTranslation example.
    """

    if (args.size != 1) {
        println(usage)
        exitProcess(0)
    }

    val jobId = args[0]
    describeTranslationJob(jobId)
}

// snippet-start:[translate.kotlin._describe_jobs.main]
suspend fun describeTranslationJob(id: String?) {

    val textTranslationJobRequest = DescribeTextTranslationJobRequest {
        jobId = id!!
    }

    TranslateClient { region = "us-west-2" }.use { translateClient ->
        val jobResponse = translateClient.describeTextTranslationJob(textTranslationJobRequest)
        println("The job status is ${jobResponse.textTranslationJobProperties?.jobStatus}.")
    }
}
// snippet-end:[translate.kotlin._describe_jobs.main]
